kornia/geometry/epipolar/_metrics.py

```diff
@@ -76,60 +136,134 @@ def sampson_epipolar_distance(
     return (out + eps).sqrt()
 
 
-def symmetrical_epipolar_distance(
+@torch.jit.script
+def sampson_epipolar_distance(
     pts1: Tensor, pts2: Tensor, Fm: Tensor, squared: bool = True, eps: float = 1e-8
 ) -> Tensor:
-    """Return symmetrical epipolar distance for correspondences given the fundamental matrix.
+    """Return Sampson distance for correspondences given the fundamental matrix.
 
     Args:
-       pts1: correspondences from the left images with shape :math:`(*, N, (2|3))`. If they are not homogeneous,
-             converted automatically.
-       pts2: correspondences from the right images with shape :math:`(*, N, (2|3))`. If they are not homogeneous,
-             converted automatically.
-       Fm: Fundamental matrices with shape :math:`(*, 3, 3)`. Called Fm to avoid ambiguity with torch.nn.functional.
-       squared: if True (default), the squared distance is returned.
-       eps: Small constant for safe sqrt.
+        pts1: correspondences from the left images with shape :math:`(*, N, (2|3))`. If they are not homogeneous,
+              converted automatically.
+        pts2: correspondences from the right images with shape :math:`(*, N, (2|3))`. If they are not homogeneous,
+              converted automatically.
+        Fm: Fundamental matrices with shape :math:`(*, 3, 3)`. Called Fm to avoid ambiguity with torch.nn.functional.
+        squared: if True (default), the squared distance is returned.
+        eps: Small constant for safe sqrt.
 
     Returns:
-        the computed Symmetrical distance with shape :math:`(*, N)`.
+        the computed Sampson distance with shape :math:`(*, N)`.
 
     """
+    if Fm.device.type == "cuda":
```

Comment:
@ducha-aiki did you manually check this ? or it's an artifact from codeflash. as soon as we know it doesn't support gpu yet

Reply:
Yes, I manually wrote this